File: source_funcs/go_transcoder/MAXIMUM_SUM_IARRI_AMONG_ROTATIONS_GIVEN_ARRAY_1/MAXIMUM_SUM_IARRI_AMONG_ROTATIONS_GIVEN_ARRAY_1_emit.go
package main

func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}

//go:export f_gold
func f_gold(arr []int, n int) int {
	var cum_sum int = 0
	for i := int(0); i < n; i++ {
		cum_sum += arr[i]
	}
	var curr_val int = 0
	for i := int(0); i < n; i++ {
		curr_val += i * arr[i]
	}
	var res int = curr_val
	for i := int(1); i < n; i++ {
		var next_val int = curr_val - (cum_sum - arr[i-1]) + arr[i-1]*(n-1)
		curr_val = next_val
		res = max(res, next_val)
	}
	return res
}

File: rust_funcs/go_transcoder/MAXIMUM_SUM_IARRI_AMONG_ROTATIONS_GIVEN_ARRAY_1/MAXIMUM_SUM_IARRI_AMONG_ROTATIONS_GIVEN_ARRAY_1_emit.rs
#![allow(dead_code)]
#![allow(unused_imports)]
#![allow(unused_variables)]
#![allow(unused_mut)]
#![allow(non_snake_case)]
#![allow(non_camel_case_types)]
#![allow(improper_ctypes_definitions)]
#[no_mangle]
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
#[no_mangle]
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
#[no_mangle]
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
#[no_mangle]
fn len(arr: &[i32]) -> usize {arr.len()}
#[no_mangle]
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}

use std::f32;

#[no_mangle]
fn f_gold(arr: [i32;2], n: i32) -> i32 {
    let mut cum_sum = 0.0;
    for i in 0..n as usize {
        cum_sum += arr[i] as f32;
    }
    let mut curr_val = 0.0;
    for i in 0..n as usize {
        curr_val += (i as f32) * (arr[i] as f32);
    }
    let mut res = curr_val as i32;
    for i in 1..n as usize {
        let next_val = curr_val - (cum_sum - arr[i - 1] as f32) + (arr[i - 1] as f32) * ((n - 1) as f32);
        curr_val = next_val;
        res = res.max(next_val as i32);
    }
    res 
}
